packages/agreement-process/test/testAgreementConsumerDocuments.ts

```diff
@@ -1,97 +1,265 @@
+      it("should succed when the requester is the consumer or producer", async () => {
+        const authData = getRandomAuthData(
+          randomArrayItem([agreement1.consumerId, agreement1.producerId])
+        );
+        const result = await agreementService.getAgreementConsumerDocument(
           agreement1.id,
           agreement1.consumerDocuments[0].id,
           authData
-        )
-      ).rejects.toThrowError(operationNotAllowed(authData.organizationId));
+        );
+
+        expect(result).toEqual(agreement1.consumerDocuments[0]);
+      });
+
+      it("should throw an agreementNotFound error when the agreement does not exist", async () => {
+        const agreementId = generateId<AgreementId>();
+        const authData = getRandomAuthData(agreement1.consumerId);
+        await addOneAgreement(getMockAgreement(), postgresDB, agreements);
+        await expect(
+          agreementService.getAgreementConsumerDocument(
+            agreementId,
+            agreement1.consumerDocuments[0].id,
+            authData
+          )
+        ).rejects.toThrowError(agreementNotFound(agreementId));
+      });
+
+      it("should throw an operationNotAllowed error when the requester is not the consumer or producer", async () => {
+        const authData = getRandomAuthData(generateId<TenantId>());
+
+        await expect(
+          agreementService.getAgreementConsumerDocument(
+            agreement1.id,
+            agreement1.consumerDocuments[0].id,
+            authData
+          )
+        ).rejects.toThrowError(operationNotAllowed(authData.organizationId));
+      });
+
+      it("should throw an agreementDocumentNotFound error when the document does not exist", async () => {
+        const authData = getRandomAuthData(agreement1.consumerId);
+        const agreementDocumentId = generateId<AgreementDocumentId>();
+        await expect(
+          agreementService.getAgreementConsumerDocument(
+            agreement1.id,
+            agreementDocumentId,
+            authData
+          )
+        ).rejects.toThrowError(
+          agreementDocumentNotFound(agreementDocumentId, agreement1.id)
+        );
+      });
     });
 
-    it("should throw an agreementDocumentNotFound error when the document does not exist", async () => {
-      const authData = getRandomAuthData(agreement1.consumerId);
-      const agreementDocumentId = generateId<AgreementDocumentId>();
-      await expect(
-        agreementService.getAgreementConsumerDocument(
-          agreement1.id,
-          agreementDocumentId,
-          authData
-        )
-      ).rejects.toThrowError(
-        agreementDocumentNotFound(agreementDocumentId, agreement1.id)
-      );
+    describe("add", () => {
+      it("should succeed on happy path", async () => {
+        const organizationId = generateId<TenantId>();
+        const authData: AuthData = {
+          ...getRandomAuthData(organizationId),
+          userRoles: ["admin"],
+        };
```

Comment:
`getRandomAuthData` already generates `AuthData` with role admin and a random organization id by default, here you could do the following 👇 (same in other tests below).

```suggestion
        const authData: AuthData = getRandomAuthData(organizationId);
        const organizationId = authData.organizationId;
```

Reply:
Yep, I thought it was better to explicit it but it makes sense to use defaults